### ISSUE TITLE
[Help Update] For v2.2.0

### ISSUE THREAD

**Issue by kiki-ja** [cleidigh/printing-tools-ng#168]
@cleidigh 
All

**[Section]**
- PTNG Overview
- Adding PTNG Button to the toolbar (New)
    Step to place the button on the toolbar (New)
- Example header formatting features (New)
- Custom Date Format
    Examples table
    Conversion Sequence Descriptions table
    Compatibility notes
- PDF Output and Auto File Naming (New)
    Examples (New)
    Token and Descriptions table
- Print and Page Options (New)
    Notes (New)
- Support
- Debug Options info (New)
- Credits (New)

**[Add and Change]**
- Help icon link to the section
- Open help in a tab
- Scale info
- Debug Option "helpinwin" to open help in a window
- Debug Options section - translations for all locales
 
**[Other]**
- Translate (Localize)
Improved en-US help
Translate all help
Updated ja locale files

- UI and Layout
Modify ja help for table hdr and "Sequence" formatting
Tweak Suffix and Prefix defs in help for de
- etc

 2023/03/09

@cleidigh
Please edit (change, modify, add, delete) the above as needed.
Of course, you may delete this issue and create a new one.

Mention #157

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

Proposal:
A uniform and aligned position of the title of the table.
For example, them aligned to the left or to the center.

Screenshots:
![help-table-header-4](https://user-images.githubusercontent.com/80428201/219944106-ab8c86b8-3aa6-42f8-914b-4ed5f725b434.png)

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

Proposal:
Support for dark mode.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Yes, we need help in dark mode.
I am going to work on help some tomorrow.
Hoping that the margins issue gets settled.
@cleidigh

**Comment by cleidigh** [comment]
@kiki-ja 
@tidebre67
@RRoenn 

I have added a lot to help and done all the localization.

@kiki-ja 
I aligned left margins and aligned table headers left. One exception,the Description.
The "Sequence" header is on two lines, not sure if this is correct?

Before any translation edits let's make such en formatting is set.
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

>I have added a lot to help and done all the localization.

I will send you printingtoolsng-help.html with corrected Japanese later.

>I aligned left margins and aligned table headers left. One exception,the Description.

Understood.

>The "Sequence" header is on two lines, not sure if this is correct?

No, please make it one line.
![Help-Sequence-width-diff](https://user-images.githubusercontent.com/80428201/220340767-298a7728-5d30-45f2-967a-bd7fac053269.png)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
I may add a section so hold off on final translation 
@cleidigh

**Comment by RRoenn** [comment]
@cleidigh
@tidebre67

I just read the new help - nice to read and good understanding

1. please edit description of prefix and suffix, what do you think?

**old**
${prefix} | Der Wert der Präfixtextoption.
${suffix} | Der Suffix-Textoptionswert.

**new**
${prefix} | Der Wert der Präfix-Textoption.
${suffix} | Der Wert der Suffix-Textoption.

2. what are ${smart_name} ? only send items?

3. please edit description, what do you think?
&D = Datum **~(beim Speichern)~**
&P = Seitenzahl **~in gespeicherter Datei~**
&PT = Seitenzahl mit Summe („3 von 5“)
&T = *~Seitentitel (~*Betreff der Nachricht~*)*~
&U = URL (Die URL der Nachricht)

&D not only the date of saving, I think also the current date while printing. Just strip it. or (= Datum (Aktuelles Datum))
&P not only in saved file ... Just strip it.
&PT add z.B. edit to (Seitenzahl mit Summe (z.B. „3 von 5“))

or short like this

    &P = Page Number (**Seitenzahl**)
    &D = Date Time (**Datum**)
    &T = Message Title (**Betreff der Nachricht**)
    &U = Message Address (Local or URL) (?? sorry ??)
    &PT = Page Number / Total number of pages (Seitenzahl mit Gesamtseitenanzahl)


@tidebre67
what do you think?


@RRoenn

**Comment by tidebre67** [comment]
> @cleidigh @tidebre67
> 
> I just read the new help - nice to read and good understanding
> 
> 1. please edit description of prefix and suffix, what do you think?
> 
> **old** ${prefix} | Der Wert der Präfixtextoption. ${suffix} | Der Suffix-Textoptionswert.
> 
> **new** ${prefix} | Der Wert der Präfix-Textoption. ${suffix} | Der Wert der Suffix-Textoption.
> 
> 2. what are ${smart_name} ? only send items?
> 3. please edit description, what do you think?
>    &D = Datum **~(beim Speichern)~**
>    &P = Seitenzahl **~in gespeicherter Datei~**
>    &PT = Seitenzahl mit Summe („3 von 5“)
>    &T = ~**Seitentitel (**~ Betreff der Nachricht ~**)**~
>    &U = URL (Die URL der Nachricht)
> 
> &D not only the date of saving, I think also the current date while printing. Just strip it. or (= Datum (Aktuelles Datum)) &P not only in saved file ... Just strip it. &PT add z.B. edit to (Seitenzahl mit Summe (z.B. „3 von 5“))
> 
> or short like this
> 
> ```
> &P = Page Number (**Seitenzahl**)
> &D = Date Time (**Datum**)
> &T = Message Title (**Betreff der Nachricht**)
> &U = Message Address (Local or URL) (?? sorry ??)
> &PT = Page Number / Total number of pages (Seitenzahl mit Gesamtseitenanzahl)
> ```
> 
> @tidebre67 what do you think?
> 
> @RRoenn

@RRoenn 
@cleidigh 
regarding prefix and suffix cannot comment, not familar with this
Page layout in favour of short description

**Comment by kiki-ja** [comment]
@cleidigh

Understood.
Please make an announcement when the editing process is completed and finished.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-b25
Thunderbird 102.8.0 64bit + Default Theme (System theme auto)
Windows 10 Pro 22H2 64bit

>One exception,the Description.

Is the description title section of the token treated as an exception?
 ![220b25-help-description- diff](https://user-images.githubusercontent.com/80428201/220931602-952b2b5e-b5d7-4158-9741-abc216794413.png)

**About the location of the help icon on tab 5(Print and Page Options):**

1. Since there is no corresponding entry for 'Default Printer' in the help, I feel it would be  better to move it to the 'Page Ranges' section.
The reason for this is that we think it would be better to display it in the same way as the other tabs, in the place of the corresponding item in the help.
 ![220b25-tab5-help-icon-1](https://user-images.githubusercontent.com/80428201/220932277-0f9a3d91-bb3a-44a1-994d-bd99aa039a71.png)

2. There is no section in the help corresponding to "Default Printer", but there is a related section "Printing without the dialog window", and in this case, I feel it would be good to place it in two places.
 ![220b25-tab5-help-icon-2](https://user-images.githubusercontent.com/80428201/220932931-9e4a4e44-57d7-49ad-8c4b-6dc97889e96e.png)

What do you think?

@kiki-ja

**Comment by RRoenn** [comment]
I vote for 2.

@RRoenn

**Comment by kiki-ja** [comment]
@cleidigh

**About the location of the help icon on tab 5(Print and Page Options):**
I would add 3.

3. The following two locations, limited to the items listed in the help section.
  "Print without the dialog window" and "Page Options",

　　　![220b25-tab5-help-icon-3](https://user-images.githubusercontent.com/80428201/221179000-bf02eef5-d737-4598-a0aa-7ede5c1fdb7a.png)

I would prefer 3.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
I made both at the section level.
I will probably add more to the help.
This also looks better.

Added button info
Help on button install 
Help opens on install or update 
Help added to button 
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh 

>I made both at the section level.
>I will probably add more to the help.
>This also looks better.

I agree.
Please add a link to the help section to all help icons.

[Example image]
![help-link-open](https://user-images.githubusercontent.com/80428201/221382862-d7c5078b-fd0c-44ad-bb42-3a6031e63883.png)

Reason for proposal:
1. It will be easier to browse directly to the relevant section without having to search in the help.
2. Due to the increased number of help items and help icons being placed.

Current Status:
Need to open Help and then look for related items.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Looking at your screenshot the top help icon is missing???
Also I fixed all links to bookmarks pointing to the specific topic ( exactly what you propose)
Something doesn't sound right.
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

The en-US version of the help has all the links added, so that is not a problem, but the ja version did not have the links yet.
Sorry, I wanted to inform you of that difference.

@kiki-ja

**Comment by cleidigh** [comment]
Ahh yes because I forgot that would also require a translation!

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-b26
Thunderbird 102.8.0 64bit + Default Dark Theme
Windows 10 Pro 22H2 64bit

Dark mode is Good.
![220b26-Help-Dark mode](https://user-images.githubusercontent.com/80428201/221541087-da11c1d5-7f4a-472e-9904-0404a5ed4e61.png)

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Help bookmarks updated in all locales.
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-b27

>Help bookmarks updated in all locales.

Thanks, Good!

@kiki-ja

**Comment by tidebre67** [comment]
@cleidigh 
don't know whether only my notebook is concerned, but help windows open in  rather large size, i.e edges of help windows 
beyond  the screen.

**Comment by kiki-ja** [comment]
@cleidigh

Suggestion: Behavior of opening help window during installation

How about making only the 'Adding the Print NG Button to the toolbar' in Help open in a slightly smaller window?

Reasons:
1. Users are surprised when the help window suddenly opens upon installation.
2. If you want to emphasize to the user the 'Adding the Print NG Button to the toolbar', it would be better to display only that area.(blue border range)
3. Make the window size a little smaller so that users are not surprised.

See screenshot.
![Adding the Print NG Button to the toolbar-0](https://user-images.githubusercontent.com/80428201/222126068-74ca535c-2976-496c-a36d-b7424ca5a38e.png)

What do you think?

@kiki-ja

**Comment by cleidigh** [comment]
@tidebre67 
@kiki-ja 
The solution I would like to go with is opening all help (install or from options ui) in a tab. This is what I always intended, but had some issue. This will not change TB size and content will be constrained.
Sound better?
<img width="960" alt="2023-03-01-12-25-18 " src="https://user-images.githubusercontent.com/25272315/222217730-94eac9d2-88a4-4d3a-90dd-130033d29235.png">

@cleidigh

**Comment by tidebre67** [comment]
@cleidigh
for me it sounds good.

**Comment by cleidigh** [comment]
All
I will edit help for scale replacement

**Comment by kiki-ja** [comment]
@cleidigh

>The solution I would like to go with is opening all help (install or from options ui) in a tab.

I do not prefer to open in tab.
The same goes for the Thunderbird option tab.
The reason is that it requires switching tabs.
If opened in a window, no switching is required.

It would be nice to be able to choose whether to open help in a tab or window.
But if there is no other way to improve it, I agree.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

PTNG 2.2.0-b28

Please again, make it one line.
![Sequence Title-Diff](https://user-images.githubusercontent.com/80428201/222437464-353f8c6c-8d98-4010-8aba-a35d02ceba16.png)

@kiki-ja

**Comment by RRoenn** [comment]
@cleidigh 

@kiki-ja 
> It would be nice to be able to choose whether to open help in a tab or window.

Unfortunately, there will be no choice, since the help is intended to be opened during the initial installation.

I prefer the tab variant. The pop-up window nearly killed me.

@RRoenn

**Comment by cleidigh** [comment]
@kiki-ja 
From a comparison to other extensions as well as how Thunderbird is moving away from separate windows, the tab makes sense.

That being said I can make a debug option winhelp that would make help open in a window. The help opened on install or update cannot because the background process for that has no access to any configuration information. Since that is a once per version event, no big deal.
I think that fits everyones needs?
@cleidigh

**Comment by cleidigh** [comment]
@kiki-ja 
I am not ignoring your point about the two line break for the Sequence header. The formatting is somehow affected, it's almost like the first character forces a new line. With other locales the first column expands to fit the content. To try and fix this I may have to do something ja styleing specific, but also I think you had a screenshot with a different translation. So we will fix that after final content: see below.

To @RRoenn and @tidebre67 also some info on how the help is done:

- I have made js scripts to use Google translation API
- These allow me to add dtd, properties, manifest and most complex, help translations 
- For help this is the process 
- The en-US file is the master
- Each time help is modified the script tells Google to translate the html page for every locale we have  (26)
- This is and has to be a one way process, everything gets written over each time, cannot do partials
- Once everything is basically in order then we can manually fix or improve a particular locale
- The problem is once you make any changes  to a locale you cannot do the script translations 
- So to keep changes you have to protect the files and manage them manually 
- Bottom line is to only translate at the last minute 
- Everything up to now were the quick, auto translations or work in progress 

Now I have to edit the help for scaling and then that becomes the new baseline 
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

>From a comparison to other extensions as well as how Thunderbird is moving away from separate windows, the tab makes sense.

I understand that Thunderbird is moving towards tabs.
However, I feel it is a matter of personal preference.
As I mentioned earlier, I don't like the idea of opening anything in tabs from an operational and work point of view.

>Since that is a once per version event, no big deal.

Yes, I think you are right.
At the very least, I wish the help that opens from the PTNG Button had a choice between tabs and windows.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
I have the ability to open the help in a window for you 👍 
it will be in b29 %-}

I am hoping I can get this out today so we can do the localization.
@cleidigh

**Comment by cleidigh** [comment]
@kiki-ja 
En help is frozen and translations done so feel free to modify the ja file.
I did get the Sequence header fixed.
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

>En help is frozen and translations done so feel free to modify the ja file.

Understood.
I will send you the corrected ja file shortly.

>I did get the Sequence header fixed.

I was testing with the same width and align as well.
Thanks for the fix.

![Sequence-Title-ja](https://user-images.githubusercontent.com/80428201/222877302-aeef977b-96e2-45da-88f3-44bf6bfd75ec.png)

@kiki-ja

**Comment by cleidigh** [comment]
@RRoenn 
@kiki-ja 
@tidebre67 
@JF1313 

b31 has the following help updates 
- Added debug options section - translations for all locales 
- merged @kiki-ja ja help 
- Updated ja locale files
- de tweak for page of pages

@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

>b31 has the following help updates
>    Added debug options section - translations for all locales

I have sent you a help file with the above added.
Please also add a link (bookmark) and help icon to the added debug options section.

![Debug-Options-Add-Help-Icon](https://user-images.githubusercontent.com/80428201/223695003-654b56d5-4116-476e-99a8-505a5a7ce000.png)

@kiki-ja

**Comment by cleidigh** [comment]
@RRoenn 
@tidebre67 
@kiki-ja 
@JF1313 

b32pre1 has the following off the list : 
- ja and de help tweaks 
- ja locale files
- fix incorrect help dirs
- printsettings debug 

- #157 
@cleidigh

**Comment by JF1313** [comment]
@Cledigh 
Clicking on the link to download the latest file says:

The 'cleidigh/printing-tools-ng' repository doesn't contain the 'xpi/printing-tools-ng-2.2.0-b32pre1-tb.xpi' path in 'v2.2.0'.  

Thanks.

Jeffrey

**Comment by cleidigh** [comment]
@JF1313 
Oops, forgot to push %-(
Fixed, thanks!
Christopher

**Comment by JF1313** [comment]
@Cledigh

All is working now! Thanks you :-)

**Comment by cleidigh** [comment]
@kiki-ja 
Did I get all the ja files merged correctly?
Does the help look good?
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

ｂ３２pre1.

>Did I get all the ja files merged correctly?

Yes, the merge has been done correctly.

>Does the help look good?

It looks great.

@kiki-ja
